src/mdlComponentHandler.js

```diff
@@ -82,42 +82,50 @@ var componentHandler = (function() {
   /**
    * Upgrades a specific element rather than all in the DOM.
    * @param {HTMLElement} element The element we wish to upgrade.
-   * @param {string} jsClass The name of the class we want to upgrade
+   * @param {string} optJsClass Optional name of the class we want to upgrade
    * the element to.
    */
-  function upgradeElementInternal(element, jsClass) {
-    if (jsClass === undefined) {
-      for (var i = 0; i < registeredComponents_.length; i++) {
-        if (element.classList.contains(registeredComponents_[i].cssClass)) {
-          upgradeElementInternal(element, registeredComponents_[i].className);
-        }
-      }
-      return;
-    }
-
+  function upgradeElementInternal(element, optJsClass) {
     // Only upgrade elements that have not already been upgraded.
     var dataUpgraded = element.getAttribute('data-upgraded');
+    var isUpgraded = function(jsClass) {
```

Comment:
I believe declaring function as variable makes it technically speaking an anonymous function, which may not be the end of the world right now, but if there are bugs later if it gets changed in future it makes it harder to debug as the console will just say anon rather than name. Could we change it to a named function eg function isUpgraded() {}
https://en.wikibooks.org/wiki/JavaScript/Anonymous_Functions